### ISSUE TITLE
Python: Parallel function calls result in event loop RuntimeErrors

### ISSUE THREAD

**Issue by sarah-widder** [microsoft/semantic-kernel#5396]
**Describe the bug**
When testing the sample chat function in `azure_chat_gpt_api.py` in a parallel setting, the calls to `kernel.invoke` all succeed, but a RuntimeError is raised about the event loop being closed.

**To Reproduce**
Modified `azure_chat_gpt_api.py` sample to call the chat function in parallel with `asyncio.gather`:

```
# Copyright (c) Microsoft. All rights reserved.

import asyncio
import logging

import semantic_kernel as sk
import semantic_kernel.connectors.ai.open_ai as sk_oai
from semantic_kernel.contents.chat_history import ChatHistory
from semantic_kernel.utils.settings import azure_openai_settings_from_dot_env_as_dict

logging.basicConfig(level=logging.WARNING)

system_message = """
You are a chat bot. Your name is Mosscap and
you have one goal: figure out what people need.
Your full name, should you need to know it, is
Splendid Speckled Mosscap. You communicate
effectively, but you tend to answer with long
flowery prose.
"""

kernel = sk.Kernel()

service_id = "chat-gpt"
chat_service = sk_oai.AzureChatCompletion(
    service_id=service_id, **azure_openai_settings_from_dot_env_as_dict(include_api_version=True)
)
kernel.add_service(chat_service)
req_settings = kernel.get_prompt_execution_settings_from_service_id(service_id=service_id)
req_settings.max_tokens = 2000
req_settings.temperature = 0.7
req_settings.top_p = 0.8
req_settings.auto_invoke_kernel_functions = True


chat_function = kernel.create_function_from_prompt(
    prompt=system_message + """{{$chat_history}}{{$user_input}}""",
    function_name="chat",
    plugin_name="chat",
    prompt_execution_settings=req_settings,
)

history = ChatHistory()
history.add_user_message("Hi there, who are you?")
history.add_assistant_message("I am Mosscap, a chat bot. I'm trying to figure out what people need.")

async def process_one_chat(user_input):
    answer = await kernel.invoke(
        chat_function,
        user_input=user_input,
        chat_history=history
    )
    print("Answer:", answer)


async def main() -> None:
    inputs = ["how are you", "what can you tell me", "what do you do", "tell me a story"]
    tasks = [process_one_chat(user_input) for user_input in inputs]
    await asyncio.gather(*tasks)

if __name__ == "__main__":
    asyncio.run(main())
```

This results in Mosscap's answers being printed, followed by 4 instances of:
```
Exception ignored in: <function _ProactorBasePipeTransport.__del__ at 0x000001959CBC6AF0>
Traceback (most recent call last):
  File "C:\Users\sawidder\AppData\Local\miniconda3\envs\sk-dev\lib\asyncio\proactor_events.py", line 116, in __del__
  File "C:\Users\sawidder\AppData\Local\miniconda3\envs\sk-dev\lib\asyncio\proactor_events.py", line 108, in close
  File "C:\Users\sawidder\AppData\Local\miniconda3\envs\sk-dev\lib\asyncio\base_events.py", line 719, in call_soon
  File "C:\Users\sawidder\AppData\Local\miniconda3\envs\sk-dev\lib\asyncio\base_events.py", line 508, in _check_closed
RuntimeError: Event loop is closed
```

**Expected behavior**
Event loop handling should properly close all async loops.

**Screenshots**
N/A

**Platform**
 - OS: Windows
 - IDE: VS Code
 - Language: Python
 - Source: pip package version 0.9.1b1  

**Additional context**
It may be helpful to add integration tests for this case :)

**Comment by moonbox3** [comment]
Hi @sarah-widder, the issue you're experiencing, `RuntimeError: Event loop is closed`, occurs in Windows when async tasks try to complete after the asyncio event loop has been closed, often during cleanup at the end of a script. A workaround is to explicitly close the event loop and create a new one before the program exits, ensuring all async operations are completed and resources are properly cleaned up, thus preventing the error.

When I add the following to the script, there's no longer an error:

```
if __name__ == "__main__":
    loop = asyncio.get_event_loop()
    try:
        loop.run_until_complete(main())
    finally:
        loop.close()
        asyncio.set_event_loop(asyncio.new_event_loop())
```

**Comment by sarah-widder** [comment]
Thanks @moonbox3 this works for me as well!

**Comment by moonbox3** [comment]
Thanks for confirming. I will close this issue -- please re-open or create a new issue, as needed.
